airflow/contrib/kubernetes/worker_configuration.py

```diff
@@ -213,8 +253,7 @@ def generate_dag_volume_mount_path(self):
     def make_pod(self, namespace, worker_uuid, pod_id, dag_id, task_id, execution_date,
                  try_number, airflow_command, kube_executor_config):
         volumes_dict, volume_mounts_dict = self.init_volumes_and_mounts()
-        worker_init_container_spec = self._get_init_containers(
-            copy.deepcopy(volume_mounts_dict))
+        worker_init_container_spec = self._get_init_containers()
```

Comment:
Why does this ignore the volumes_dict returned form `init_volumes_and_mounts`?

Reply:
There are three volumes: the dags volume, the logs volume and the git ssh secret volume (if SSH is turned on)

with the deep copy you had the init container having all of the mounts that the pod container has, but there is no logical need for the init container:
- to share the same mount path for the dags (and we can manage security for the init pod better)
 - no needs for the Airflow logs to be mounted
 - the git SSH secret can be mounted to the init container without it being mounted on the worker pods (so you only have it on the worker during initialization, no need for SSH github secret in the worker post initialization since the DAGs are pulled once)

Reply:
Ahhhh gotcha. Nice